hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/Sinks.java

```diff
@@ -675,7 +681,57 @@ private Sinks() {
      * @param <T> type of the items the sink accepts
      */
     @Nonnull
-    public static <T> JmsSinkBuilder<T> jmsTopicBuilder(DistributedSupplier<ConnectionFactory> factorySupplier) {
-        return new  JmsSinkBuilder<>(factorySupplier, true);
+    public static <T> JmsSinkBuilder<T> jmsTopicBuilder(@Nonnull DistributedSupplier<ConnectionFactory> factorySupplier) {
+        return new JmsSinkBuilder<>(factorySupplier, true);
+    }
+
+    /**
+     * Returns a sink that connects to the specified database using the given
+     * {@code connectionSupplier} and writes to it using the given {@code
+     * statementFn} and {@code updateFn}. After a batch of items is written,
+     * {@code flushFn} is called to flush the changes.
+     * <p>
+     * No state is saved to snapshot for this sink. After the job is restarted,
+     * the items will likely be duplicated, providing an <i>at-least-once</i>
+     * guarantee.
+     * <p>
+     * The default local parallelism for this sink is 1.
+     * @param connectionSupplier the supplier of database connection
+     * @param statementFn the function to prepare the statement
+     * @param updateFn the function to set the parameters of the statement for
+     *                 each item received and execute the update query. Instead
+     *                 of updating for each item, it can be used to add to batch
+     *                 via {@link PreparedStatement#addBatch()}
+     * @param flushFn the function to flush the changes. Gets {@code connection}
+     *                and {@code statement} as arguments. It can be used to
+     *                commit the connection if it is not {@code autoCommit} mode
+     *                or execute the batch
+     * @param <T> type of the items the sink accepts
+     */
+    @Nonnull
+    public static <T> Sink<T> jdbc(
+            @Nonnull DistributedSupplier<Connection> connectionSupplier,
+            @Nonnull DistributedFunction<Connection, PreparedStatement> statementFn,
+            @Nonnull DistributedBiConsumer<PreparedStatement, T> updateFn,
+            @Nonnull DistributedBiConsumer<Connection, PreparedStatement> flushFn
+    ) {
+        return Sinks.fromProcessor("jdbcSink",
+                SinkProcessors.writeJdbcP(connectionSupplier, statementFn, updateFn, flushFn));
+    }
+
+    /**
+     * Convenience for {@link Sinks#jdbc(DistributedSupplier,
+     * DistributedFunction, DistributedBiConsumer, DistributedBiConsumer)}.
+     */
+    @Nonnull
+    public static <T> Sink<T> jdbc(@Nonnull String connectionUrl,
+                                   @Nonnull String updateQuery,
+                                   @Nonnull DistributedBiConsumer<PreparedStatement, T> updateFn) {
```

Comment:
Here we also require the connection to be in auto-commit mode because we don't commit. We can add 
```
if (!con.getAutoCommit()) {
    con.commit();
}
```
to the `flushFn`.

Reply:
from javadoc `By default a <code>Connection</code> object is in auto-commit mode...`

Reply:
user cannot change it, this is convenience method, we do create the connection using the connectionURL.
`() -> uncheckCall(() -> DriverManager.getConnection(connectionUrl)),`

Reply:
ah, true. Then i'll disable auto-commit and i'll commit in flushFn. It's much more performant.